### PR TITLE
feat: add milticall contract

### PR DESCRIPTION
**Pull request by SlayerAnsh**
# Pull Request

## Description

This pull request introduces the MultiCall contract, which allows for batch processing of multiple queries in a single transaction. This enhancement improves efficiency and reduces the overhead associated with multiple individual calls.

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update
- [ ] Refactor/Chore

## Queries Implemented
- MultiQuery: Allows users to submit multiple queries in a single transaction.

### Query Type
- Query: Supports standard query requests using QueryRequest<Empty>.
- RawQuery: Enables execution of raw string queries for more flexibility.

Sample JSON
```json
{
  "queries": [
    {
      "bank": {
        "balance": {
          "address": "creator",
          "denom": "test"
        }
      }
    },
    {
      "raw_query": "{\"bank\":{\"balance\":{\"address\":\"creator\",\"denom\":\"test\"}}}"
    }
  ]
}
```

## Related Issue

N/A

## Testing

Steps to test:

1. Deploy the MultiCall contract.
2. Execute multiple queries using the MultiQuery feature.
3. Verify that all queries return the expected results in a single response

## Checklist

- [x] Self-review completed
- [x] Tests added/updated (if applicable)
- [ ] Documentation updated (if applicable)

## Screenshots (if applicable)

N/A

## Additional Notes

This implementation is designed to streamline interactions with the contract, making it easier for users to perform multiple queries efficiently.

### CODE REVIEW

**Comment by SlayerAnsh** [comment]
Got successful result with on chain testing for queries

File: contracts/common/cw-multicall/src/contract.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+use euclid::error::ContractError;
+
+use crate::query::query_multi_queries;
+use cw2::set_contract_version;
+
+use euclid_utils::msgs::multicall::{ExecuteMsg, InstantiateMsg, QueryMsg};
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:cw-multicall";
```
Comment (joemonem) [inline contracts/common/cw-multicall/src/contract.rs:13]: should we call this multi-call?

**Comment by joemonem** [comment]
We should map out our current workflows that could benefit from this contract and integrate them.

File: contracts/common/cw-multicall/src/contract.rs (hunk 1 above)
Comment (SlayerAnsh) [reply contracts/common/cw-multicall/src/contract.rs:13]: We can but this has been added as multicall in our api so lets keep it `multicall` now

**Comment by SlayerAnsh** [comment]
> We should map out our current workflows that could benefit from this contract and integrate them.

Work has been started to make composite graphql queries using multicall to minimize calls to chain nodes

**Review by joemonem — APPROVED** [review]

**Pull request merged**